crates/muvm/src/guest/server.rs

```diff
@@ -84,14 +88,13 @@ pub async fn server_main(command: PathBuf, command_args: Vec<String>) -> Result<
                         );
                     },
                 }
-                command_exited = true;
             },
-            Some(state) = state_rx.next(), if command_exited => {
+            Some(state) = state_rx.next(), if command_exited_code.is_some() => {
                 if state.connection_idle() && state.child_processes() == 0 {
                     // Server is idle (not currently handling an accepted
                     // incoming connection) and no more child processes.
                     // We're done.
-                    return Ok(());
+                    std::process::exit(command_exited_code.unwrap());
```

Comment:
I suspect this is really bad. We should generally not call `std::process::exit` (which causes an abrupt exit), but bubble up the exit code properly.

The docs has ample warning and guidance: 

https://doc.rust-lang.org/std/process/fn.exit.html